src/mapeo-project.js

```diff
@@ -749,6 +761,410 @@ export class MapeoProject extends TypedEmitter {
+   * @param {Object} options
+   * @param {Set<string>} [options.seenObservations=new Set()]
+   * @param {SeenAttachments} [options.seenAttachments]
+   * @returns {AsyncIterable<Buffer | Uint8Array>}
+   */
+  async *#exportObservations(
+    observations,
+    { seenObservations = new Set(), seenAttachments = new Map() }
+  ) {
+    let first = true
+    for (const observation of observations) {
+      const { lat, lon, docId } = observation
+      if (seenObservations.has(docId)) {
+        continue
+      }
+      seenObservations.add(docId)
+      for (const attachment of observation.attachments) {
+        const { hash } = attachment
+        if (!seenAttachments.has(hash)) {
+          seenAttachments.set(hash, attachment)
+        }
+      }
+
+      let latitude = lat
+      let longitude = lon
+      let altitude = null
+      const position = observation?.metadata?.position?.coords
+      if (position) {
+        latitude = position.latitude
+        longitude = position.longitude
+        if (position.altitude !== undefined) {
+          altitude = position.altitude
+        }
+      }
+
+      const coordinates = [longitude, latitude]
+      if (typeof altitude === 'number') {
+        coordinates.push(altitude)
+      }
+      const hasLatLon =
+        typeof longitude === 'number' && typeof latitude === 'number'
+      const geometry = hasLatLon
+        ? {
+            type: 'Point',
+            coordinates,
+          }
+        : null
+      const comma = first ? '' : ','
+      first = false
+      yield b4a.from(
+        `${comma}\n      ` +
+          JSON.stringify({
+            type: 'Feature',
+            properties: observation,
+            geometry,
+          })
+      )
+    }
+  }
+
+  /**
+   * @param {Iterable<Track>} tracks
+   * @param {Object} options
+   * @param {Set<string>} [options.seenObservations=new Set()]
+   * @param {SeenAttachments} [options.seenAttachments]
+   * @param {string} [options.lang]
+   * @returns {AsyncIterable<Buffer | Uint8Array>}
+   */
+  async *#exportTracks(
+    tracks,
+    { lang, seenObservations = new Set(), seenAttachments = new Map() } = {}
+  ) {
+    let first = true
+    for (const track of tracks) {
+      const { observationRefs } = track
+
+      const observations = await Promise.all(
+        observationRefs.map(({ docId }) =>
+          this.#dataTypes.observation.getByDocId(docId, { lang })
+        )
+      )
+
+      const coordinates = track.locations.map(
+        ({ coords: { longitude, latitude, altitude } }) => [
+          longitude,
+          latitude,
+          altitude,
+        ]
+      )
+      const comma = first ? '' : ','
+      first = false
+      yield b4a.from(
+        `${comma}\n      ` +
+          JSON.stringify({
+            type: 'Feature',
+            properties: track,
+            geometry: {
+              type: 'LineString',
+              coordinates,
+            },
+          }) +
```

Comment:
It's unfortunately that there is no standard JSON format for storing timestamps or other event data for each position in a track. It might be useful to add as a possible follow-up feature storing the start and end datetime for a track, since we can't export the timestamps for each position. The only export format that supports this I think is GPX (an XML format). I found this old discussion https://gis.stackexchange.com/questions/162081/representing-time-in-geojson and I remember other attempts to address this in the past, but none came to anything.